src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java

```diff
@@ -78,16 +83,26 @@ public void parseAndStoreTranscript(String videoId, Key lectureKey) throws IOExc
 
   /**
    * Returns the transcript for a video as a document. Otherwise, returns Optional.empty()
-   * if there is a parsing error.
+   * if there is a parsing or URL building error.
    *
    * @param videoId Indicates the video to extract the transcript from.
    */
   private Optional<Document> getTranscriptXmlAsDocument(String videoId) throws IOException {
-    String transcriptXMLUrl = XML_URL_TEMPLATE + videoId;
+    final URL url;
```

Comment:
nit: would it be possible to move the URL building code to its own method?

Reply:
Done, take another look